pyani_plus/db_orm.py

```diff
@@ -88,6 +96,61 @@ def __repr__(self) -> str:
         )
 
 
+class Configuration(Base):
+    """Describes the configuration of a pyANI-plus run.
+
+    Each run is a set of all-vs-all comparisons between a set of FASTA
+    files (genome table rows), for a specific ANI algorithm or method,
+    recorded as a set of comparison table rows.
+
+    Each Run entry represents a specific instance of an analysis run which
+    records details like the exact command line, and when it was run. It will
+    point to a single Configuration entry (this class) which records the tool
+    name, version, and parameters like k-mer size (some parameters are tool
+    specific).
+
+    Comparison entries are cached for unique combinations of the query and
+    subject genomes (via the hash of their file contents, not via filenames),
+    AND the run configuration.
+    """
+
+    __tablename__ = "configuration"
+
+    __table_args__ = (
```

Comment:
I've added two fields, (operating) system aka ``uname -s`` and machine aka ``uname -m`` aka ``arch`` for the CPU architecture.

I am unsure how much importance to place on the operating system revision (``uname -r``), although adding it will likely not be an issue for most cluster usage as we would expect all the nodes in an HPC to run the same version of the OS.

For reference, my M3 apple laptop reports:

```
$ uname -srm
Darwin 23.6.0 arm64
```